### DISCUSSION TITLE
OpenVPN auth-user-path Not Working

### DISCUSSION THREAD

**Discussion by GitHubUser655321** [RaspAP/raspap-webgui#1301] [General]
I'm using OpenVPN in RaspAP 2.8.7 on an Orange Pi 3.0.8 running Ubuntu Jammy 22.04 with Linux 5.16.17-sun50iw6.

When I enter my username and password for my VPNSecure account and attach and upload the conf file, I don't get any errors and it does say it added the auth-user-path directive successfully. I check the files on the server and the ...client.conf has the auth-user-path pointed to the login.conf file and the ...login conf file does contain the correct credentials.

The problem seems to be that OpenVPN isn't reading the login.conf file.
When I try to connect it times-out waiting for a password. No other errors in the log.
If I have a terminal connection to the server open, I can use systemd-tty-ask-password-agent to enter my password and it does make a successful connection.

I have tried a few things to troubleshoot the issue;
I've changed the permissions on the file to 777 just to make sure it can be read.
I've entered the full path to the file, to make sure it knows where it is.
I've removed the `<cert>` from the file to make sure there wasn't a conflict with that.

I'm hoping someone could give me a few more ideas to troubleshoot the issue.

Here's my conf file;

```
client
proto udp
dev tun
remote my.wicked-cool.vpn 1191
cipher DES-CBC
verb 3
mute 20
keepalive 10 120
comp-lzo
float
persist-key
persist-tun
resolv-retry infinite
nobind
auth-nocache
remote-cert-tls server
<ca>ca is here</ca>
<key>key is here</key>
<cert>cert is here</cert>
auth-user-pass login.conf
```

Thanks for any help.

**Comment by billz** [comment]
See this FAQ for troubleshooting tips: https://docs.raspap.com/faq/#openvpn-fails-to-start-andor-i-have-no-internet

**Comment by GitHubUser655321** [comment]
Thanks billz. I figured it out. 

After searching more on the web, I found an old 2017 [post on stackexchange](https://stackoverflow.com/questions/11240184/pass-private-key-password-to-openvpn-command-directly-in-ubuntu-10-10) that lead me in the right direction.

Turns out the VPN Sever was looking for just the password, not username and password. I had tried putting just the password into the _login.conf file, but the `auth-user-pass` was expecting 2 parameters. I had to change that to `askpass` for it to work.

**So, I 
manually change the `auth-user-pass` directive to `askpass` in the _client.conf 
and then 
put _only the password_ in the _login.conf file.** 
After that it worked without prompting for a password.

The only issue I see now is that the RasAP VPN interface shows my password in the Username field. No big deal I guess. And, I can change the name of the _login.conf file and change it in the _client.conf file so it doesn't do that.

Guess this will all be moot once I get a VPN service that supports WireGuard.

Thanks again!
